api/routes/user.js

```diff
@@ -0,0 +1,186 @@
+const express = require("express");
+const router = express.Router();
+const validate = require("express-jsonschema").validate;
+
+const UserSchema = require("../public/schema/userSchema.js").userSchema;
+const SUCCESS = 200;
+const NOT_FOUND = 404;
+
+router.get("/", async (req, res) => {
+  const { db } = req;
+  const collection = db.get("users");
+  const users = await collection.find({});
+  res.status(SUCCESS).send({
+    code: SUCCESS,
+    success: true,
+    message: "Users retrieved successfully.",
+    data: users
+  });
+});
+
+router.get("/:id", async (req, res) => {
+  const { db } = req;
+  const collection = db.get("users");
+  const { id } = req.params;
+  const users = await collection.find({ _id: id });
+  const ret =
+    users.length === 1
+      ? {
+          code: SUCCESS,
+          success: true,
+          message: "User retrieved successfully.",
+          data: users[0]
+        }
+      : {
+          code: NOT_FOUND,
+          success: false,
+          message: "User not found."
+        };
+  res.status(ret.code).send(ret);
+});
+
+router.post("/", validate({ body: UserSchema }), async (req, res) => {
+  const { db } = req;
+  const collection = db.get("users");
+  const { email, username, password, country, birthday, role } = req.body;
+  let { anon } = req.body;
+  if (!email) {
```

Comment:
Great work breaking down the responses for all the missing fields!!

Reply:
Thanks :D